attendance/test/userdata_test.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:attendance/BackEnd/user.dart';
+import 'package:test/test.dart';
+
+void main() {
+  User testUser = User.instance();
+  String perfectName = 'someName';
+  String perfectPhone = '1234567890';
+
+  String jsonBody =
+      '{' + '"nativeName":"$perfectName",' + '"phone":"$perfectPhone"' + '}';
+
+  test('user object init correctly', () {
+    expect(testUser, isNotNull);
+  });
+
+  group('Rename', () {
+    test('should has no default values', () {
+      testUser.rename('');
+
+      expect(testUser.nativeName, isEmpty);
+    });
+
+    test('should be matched ', () {
+      testUser.rename(perfectName);
+
+      expect(testUser.nativeName, 'someName');
+    });
+
+    test('should be no longer than 20 chars', () {
```

Comment:
@Bahaaib those validations should be inside the user class and then those cases should be tested
It's okay to leave a puzzle for it (this PR's size is nice, no need to be bigger or smaller)

Reply:
Shouldn't it be tested here too to assure of the class code quality?

Reply:
@Bahaaib yes, should make a user with invalid data and assert that it throws an error and refuse to create the user

Reply:
@Bahaaib keep those tests, will be the guide during the puzzle solving

Reply:
You mean reverse the tests?

Reply:
@Bahaaib no just add the puzzle to validate those cases in the user class